packages/mask/src/plugins/Collectible/hooks/useCollectibleState.ts

```diff
@@ -8,9 +8,14 @@ function useCollectibleState(token?: CollectibleToken) {
     const [tabIndex, setTabIndex] = useState(CollectibleTab.ARTICLE)
     const [provider, setProvider] = useState(token?.provider ?? SourceType.OpenSea)
 
-    const asset = useNonFungibleAsset(NetworkPluginID.PLUGIN_EVM, token?.contractAddress ?? '', token?.tokenId ?? '', {
-        sourceType: provider,
-    })
+    const asset = useNonFungibleAsset(
+        token?.provider === SourceType.MagicEden ? NetworkPluginID.PLUGIN_SOLANA : NetworkPluginID.PLUGIN_EVM,
```

Comment:
@guanbinrui We might consider adding a `pluginId` field to `Token`

Reply:
Not quite sure how to address this atm, any suggestions?

Reply:
It's able to lock the plugin id of a small region of context.

```tsx
<PluginIDContextProvider value={NetworkPluginId.PLUGIN_EVM}>
    <!-- usePluginID will be fixed to NetworkPluginId.PLUGIN_EVM -->
</PluginIDContextProvider>
```